### ISSUE TITLE
Finetune WMT19 en-de single model

### ISSUE THREAD

**Issue by babangain** [pytorch/fairseq#2639]
Can someone give me steps to finetune wmt19 en-de single model on custom data?

Here are the steps I have followed.

1. Downloaded the model and bpecodes
2. Preprocced the custom data using the same dictionary downloaded with model.
3. 

```bash
nohup fairseq-train --fp16 \
    data-bin/chat_data \
    --source-lang en --target-lang de \
    --arch transformer_wmt_en_de_big --share-all-embeddings --log-interval  1  --log-format simple --tensorboard-logdir tensorboard\
    --dropout 0.3 --weight-decay 0.0 \
    --criterion label_smoothed_cross_entropy --label-smoothing 0.1 \
    --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0 \
    --lr 0.0001 --lr-scheduler inverse_sqrt --warmup-updates 4000 \
    --max-tokens 896  \
    --max-epoch 3\
    --restore-file data-bin/wmt19.en-de.joined-dict.single_model/model.pt \
    --save-dir data-bin/wmt19.en-de.joined-dict.single_model &
```

It says "please ensure that the architectures match."

**Comment by scheiblr** [comment]
@babangain, that might be related to an issue I opened some time ago (https://github.com/pytorch/fairseq/issues/2553)

**Comment by babangain** [comment]
Yes.
transformer_wmt_en_de_big encoder ffn is 4096 by default and in wmt'19, authors used 8192. Changed in on transformer.py and it worked.

**Comment by scheiblr** [comment]
What did you exactly change? Could you pls share that code?

**Comment by edunov** [comment]
If you just add  --encoder-ffn-embed-dim 8192 to the fairseq-train command that should be enough.

**Comment by scheiblr** [comment]
Thanks a lot! That worked for me!
